### DISCUSSION TITLE
Experimental Live Reloading

### DISCUSSION THREAD

**Discussion by mehalter** [AstroNvim/AstroNvim#862] [Announcements]
We have just merged a new experimental feature into AstroNvim nightly that adds a new command `:AstroReload`. This attempts to do live reloading of the configuration and new changes to your user configuration without needing to restart. We have noticed some instability with this in regards to some language servers that do not correctly unattach and reattach appropriately. If you experience any instability after a reload simply restart AstroNvim.

This functionality has also added a couple new features into the AstroNvim updater to help streamline the update process. Specifically we have added two new options to the `updater` configuration: `auto_reload` and `auto_quit`. When `auto_reload = true` AstroNvim will reload the user configuration and sync Packer to get any new packages/versions and then `auto_quit` will make AstroNvim quit automatically after a successful update. We would recommend when using these features to use them in tandem. That way after running `:AstroUpdate` you simply just need to run `nvim` again to get into the fully updated AstroNvim environment without the need of running `:PackerSync` and then restarting a second time. These new options are explained in detail in the `nightly` documentation: https://astronvim.github.io/nightly/Configuration/updater#configuration

These new `auto_reload` and `auto_quit` features also pair nicely with the load last session features of AstroNvim. After reloading AstroNvim after an update running `<leader>Sl` should get you back into your previous session.

**Comment by snewell92** [comment]
I endorse this with [real cash money](https://Ko-fi.com/home/coffeeshop?txid=46775b54-ac51-4fdc-856d-3c2e45abcc9e&mode=public&img=ogiboughtsomeone)
